### PR TITLE
feature: subscriptions api

### PR DESCRIPTION
**Pull request by justinemmanuelmercado**

### CODE REVIEW

**Comment by justinemmanuelmercado** [comment]
@moltar I messed up, was looking to request a review but accidentally merged it to master instead, reverted it using the revert option. I'll revert the revert when you approve this PR

File: src/http.ts [hunk 1]
```diff
@@ -58,16 +58,25 @@ export interface MWSOptions {
...
 
 type HttpMethod = 'GET' | 'POST'
-type ParameterTypes = string | number | (number | string)[] | object[] | boolean | undefined
+type ParameterTypes =
+  | string
+  | number
+  | (number | string)[]
+  | object[]
+  | boolean
+  | { [key: string]: ParameterTypes }
```
Comment (moltar) [inline src/http.ts:67]: Better expressed as `Record<string, ParameterTypes>`

File: test/unit/http.test.ts [hunk 2]
```diff
@@ -174,5 +174,75 @@ describe('httpClient', () => {
...
+      expect.hasAssertions()
+
+      const parameters = {
+        MarketplaceId: 'ATVPDKIKX0DER',
+        Destination: {
+          DeliveryChannel: 'SQS',
+          'AttributeList.member': [
+            {
+              Key: 'sqsQueueUrl',
+              // eslint-disable-next-line sonarjs/no-duplicate-string
```
Comment (moltar) [inline test/unit/http.test.ts:188]: Let's not disable this linting rule, but instead put the value in a constant at the top of the file or relevant `describe` code block.

File: src/http.ts (hunk 1 above)
Comment (justinemmanuelmercado) [reply src/http.ts:67]: I tried that at first, but it threw `Type alias 'ParameterTypes' circularly references itself.` and only used the above as a fix. Any reason why `{ [key: string]: ParameterTypes }` won't work?

This is where I got the "fix" https://github.com/microsoft/TypeScript/issues/14174. I didn't read through everything though so there may be something I missed in that issue that might've been a better fix

File: test/unit/http.test.ts (hunk 2 above)
Comment (justinemmanuelmercado) [reply test/unit/http.test.ts:188]: Ohh you're right. I missed this, I used a quick fix with the intention of fixing it after making sure that my tests passed but forgot to go back to it

**Pull request merged**